tools/build_script_generator/cmake/module.lb

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017-2018, Fabian Greif
+# Copyright (c) 2017-2018, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+import os
+
+
+def init(module):
+    module.name = "build.cmake"
+    module.description = """CMake Build Script Generator
+
+Generates a CMakeLists.txt file to build the modm library.
+"""
+
+
+def prepare(module, options):
+    _, default_project_name = os.path.split(os.getcwd())
+
+    module.add_option(
+        BooleanOption(name="include_demo_project", default=False,
+                      description="Include a demo main file to get a full compilable project."))
+    module.add_option(
+        BooleanOption(name="include_sconstruct", default=True,
+                      description="Generate modm specific SConstruct file."))
+    module.add_option(
+        BooleanOption(name="is_unittest", default=False,
+                      description="Generate SConstruct file for modm unittests."))
+
+    module.add_option(
+        StringOption(name="build.path", default=".",
+                     description="Path to the build folder"))
+    module.add_option(
+        StringOption(name="project.name", default=default_project_name.replace('.', '_'),
+                     description="Project name (required)"))
+    module.add_option(
+        StringOption(name="image.source", default="",
+                     description="Path to the image folder"))
+    module.add_option(
+        StringOption(name="modm.source", default="",
+                     description="Path to the Modm source file folder"))
+    module.add_option(
+        StringOption(name="modm.container", default="",
+                     description="Name of the Modm container to generate for"))
+    module.add_option(
+        StringOption(name="modm.generate", default="communication",
+                     description="Path to the Modm generated folder"))
+    module.add_option(
+        StringOption(name="modm.namespace", default="robot",
+                     description="Namespace of the generated Modm communications"))
+
+    module.add_option(
+        BooleanOption(name="info.git", default=False,
+                      description="Generate CPP defines about the git repository state"))
+    module.add_option(
+        BooleanOption(name="info.build", default=False,
+                      description="Generate CPP defines about the build state"))
+
+    if options[":target"].identifier["platform"] == "avr":
+        # we need the clock:f_cpu option!
+        module.depends(":platform:clock")
+    return True
```

Comment:
Since your script only works with STM32 for the moment, use this to enable this module only for STM32 targets.
And remove the AVR stuff above.

```py
if options[":target"].identifier["platform"] == "stm32":
    return True
return False
```

Reply:
Done.